### PR TITLE
feat: provide support for .env files injects via API

### LINKED ISSUE #975: feat: provide support for .env files injects via API
relates #974 

Introduce `.env` loading via API

**./path/prod.env**
```ini
FOO=BAR
```

**my-script.mjs**
```js
$.env = './path/prod.env'
await $`echo $FOO`.stdout // BAR
```

### PR DESCRIPTION
**Pull request by easymikey**
Fixes #975

```js
/// './path/prod.env'
FOO=BAR

---
/// index.js
$.env = './path/prod.env'
await $`echo $FOO`.stdout // BAR
```

- [x] Tests pass
- [ ] Appropriate changes to README are included in PR

### CODE REVIEW

**Comment by easymikey** [comment]
Screenshot 2024-12-25 at 18 04 44

**Comment by easymikey** [comment]
Screenshot 2024-12-25 at 18 05 07

File: test/core.test.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ import { test, describe, before, after, it } from 'node:test'
 import { inspect } from 'node:util'
 import { basename } from 'node:path'
 import { WriteStream } from 'node:fs'
+import '../build/globals.js'
```
Comment (antongolub) [inline test/core.test.js:20]: Let's pick required methods explicitly

**Comment by antongolub** [comment]
What's about this way instead?

```js
const loadDotenv = (...files: string[]) => files.reverse().reduce<env>((m, f) => Object.assign(m, parseDotenv(fs.readFileSync(f))), {})

$.env = loadDotenv('.env', '.env.default')
```

**Comment by easymikey** [comment]
> What's about this way instead?
> 
> ```js
> const loadDotenv = (...files: string[]) => files.reverse().reduce<env>((m, f) => Object.assign(m, parseDotenv(fs.readFileSync(f))), {})
> 
> $.env = loadDotenv('.env', '.env.default')
> ```


Maybe replace it with spread? 
```js
const loadDotenv = (...files: string[]) =>
  files
    .reverse()
    .reduce<env>((m, f) => ({ ...m, ...parseDotenv(fs.readFileSync(f)) }), {});

```

And why the reverse? 
```js
$.env = loadDotenv('.env.default', '.env')
```

Are you suggesting we consider an array of strings as well?

```js
type Env = NodeJs.Process | string | string[]
```

**Comment by antongolub** [comment]
I suggest keeping the `$.env` contract as is, and just export `loadDotenv` helper

File: test/core.test.js (hunk 1 above)
Comment (easymikey) [reply test/core.test.js]: d1

**Review by antongolub — APPROVED** [review]
lgtm. Thanks.

**Pull request merged**